test/util/conditional.spec.js

```diff
@@ -0,0 +1,157 @@
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(false);
+    });
+
+    it('Testing eval "LessThanOrEqualTo"', () => {
+      expect(evaluate({ subject: 'id', type: 'LessThanOrEqualTo', object: '56629a41-b24b-42c6-9f02-277412e96d25' }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "GreaterThan"', () => {
+      expect(evaluate({ subject: 'id', type: 'GreaterThan', object: '56629a41-b24b-42c6-9f02-277412e96d25' }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(false);
+    });
+
+    it('Testing eval "GreaterThanOrEqualTo"', () => {
+      expect(evaluate({ subject: 'id', type: 'GreaterThanOrEqualTo', object: '56629a41-b24b-42c6-9f02-277412e96d25' }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Between"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'Between',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        lowerBound: '56629a41-b24b-42c6-9f02-277412e96d25',
+        upperBound: 'id'
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25',
+        subject: 'id'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Membership"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'Membership',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        values: [
+          'id'
+        ]
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25',
+        subject: 'id'
+      })).to.equal(false);
+    });
+
+    it('Testing eval "Function"', () => {
+      expect(() => evaluate({ subject: 'id', type: 'Function', object: '56629a41-b24b-42c6-9f02-277412e96d25' }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.throw('Not implemented');
+    });
+
+    it('Testing eval "Not"', () => {
+      expect(evaluate({
+        subject: 'id', type: 'Not', object: '56629a41-b24b-42c6-9f02-277412e96d25', condition: { type: 'Equals' }
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(false);
+    });
+
+    it('Testing eval "And"', () => {
+      expect(evaluate({
+        subject: 'id', type: 'And', object: '56629a41-b24b-42c6-9f02-277412e96d25', conditions: [{ type: 'Equals' }]
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Or"', () => {
+      expect(evaluate({
+        subject: 'id', type: 'Or', object: '56629a41-b24b-42c6-9f02-277412e96d25', conditions: [{ type: 'Equals' }]
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Default"', () => {
+      expect(() => evaluate({ subject: 'id', type: 'Default', object: '56629a41-b24b-42c6-9f02-277412e96d25' }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.throw('Unknown condition type "Default" provided');
```

Comment:
We agreed to not use `to.throw` because it can have false positives. Use the exported errors from the `errors.js` file after they've been defined.